sqlx-core/src/query_builder.rs

```diff
@@ -523,7 +533,7 @@ where
 
     /// Deconstruct this `QueryBuilder`, returning the built SQL. May not be syntactically correct.
     pub fn into_sql(self) -> String {
-        self.query
+        Arc::into_inner(self.query).unwrap()
```

Comment:
This shouldn't panic if a strong-ref is held somewhere:

```suggestion
        Arc::unwrap_or_clone(self.query)
```

Reply:
Alternatively, this could change to return `SqlStr`, which shouldn't have to copy the string.